File: patient-monitor-csharp/PatientMonitor.Api/Controllers/PatientController.cs
using Microsoft.AspNetCore.Mvc;
using PatientMonitor.Api.Dtos;
using PatientMonitor.Api.Services;

namespace PatientMonitor.Api.Controllers;

[ApiController]
[Route("api/patients")]
public class PatientController : ControllerBase
{
    private readonly IPatientService _patientService;

    public PatientController(IPatientService patientService)
    {
        _patientService = patientService;
    }

    /// <summary>获取所有患者</summary>
    [HttpGet]
    public ActionResult<List<object>> GetAllPatients()
    {
        return Ok(_patientService.GetAllPatients());
    }

    /// <summary>获取单个患者及其行为日志</summary>
    [HttpGet("{id}")]
    public ActionResult<object> GetPatient(long id)
    {
        return Ok(_patientService.GetPatientWithLogsById(id));
    }

    /// <summary>获取异常患者及其日志</summary>
    [HttpGet("abnormal")]
    public ActionResult<List<object>> GetAbnormalPatients()
    {
        return Ok(_patientService.GetAbnormalPatientsWithLogs());
    }

    /// <summary>更新患者状态</summary>
    [HttpPut("{id}/status")]
    public ActionResult<string> UpdatePatientStatus(long id, [FromBody] StatusUpdateRequest request)
    {
        _patientService.UpdatePatientStatus(id, request.Status);
        return Ok("状态更新成功");
    }

    /// <summary>更新患者异常等级</summary>
    [HttpPut("{id}/severity")]
    public ActionResult<string> UpdatePatientSeverity(long id, [FromBody] SeverityUpdateRequest request)
    {
        _patientService.UpdatePatientSeverity(id, request.Severity);
        return Ok("异常等级更新成功");
    }
}

File: patient-monitor/src/main/java/com/hospital/monitor/controller/PatientController.java
package com.hospital.monitor.controller;

import com.hospital.monitor.dto.SeverityUpdateRequest;
import com.hospital.monitor.dto.StatusUpdateRequest;
import com.hospital.monitor.service.PatientService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/patients")
@RequiredArgsConstructor
@Tag(name = "患者管理", description = "患者相关接口")
public class PatientController {

    private final PatientService patientService;

    @GetMapping
    @Operation(summary = "获取所有患者")
    public Object getAllPatients() {
        return patientService.getAllPatients();
    }

    @GetMapping("/{id}")
    @Operation(summary = "获取单个患者")
    public Object getPatient(@PathVariable Long id) {
        return patientService.getPatientWithLogsById(id);
    }

    @GetMapping("/abnormal")
    @Operation(summary = "获取异常患者及其日志")
    public Object getAbnormalPatients() {
        return patientService.getAbnormalPatientsWithLogs();
    }

    @PutMapping("/{id}/status")
    @Operation(summary = "更新患者状态")
    public Object updatePatientStatus(@PathVariable Long id,
                                       @Valid @RequestBody StatusUpdateRequest request) {
        patientService.updatePatientStatus(id, request.getStatus());
        return "状态更新成功";
    }

    @PutMapping("/{id}/severity")
    @Operation(summary = "更新患者异常等级")
    public Object updatePatientSeverity(@PathVariable Long id,
                                         @Valid @RequestBody SeverityUpdateRequest request) {
        patientService.updatePatientSeverity(id, request.getSeverity());
        return "异常等级更新成功";
    }

    @GetMapping("/{id}/behavior-logs")
    @Operation(summary = "获取患者所有行为记录（用于图表统计）")
    public Object getPatientAllBehaviorLogs(@PathVariable Long id) {
        return patientService.getAllBehaviorLogsByPatientId(id);
    }
}
